projects/plutus/inc/current_sense.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+// Current sense module for the LTC2484 ADC
+
+#include "ltc_adc.h"
+#include "status.h"
+
+typedef void (*CurrentSenseCallback)(int32_t current, void *context);
+
+typedef struct {
+  int32_t voltage;
+  int32_t current;
+} CurrentSenseValue;
+
+// User-defined points for two-point calibration
+typedef struct {
+  CurrentSenseValue zero_point;
+  CurrentSenseValue max_point;
+} CurrentSenseCalibrationData;
+
+typedef struct {
+  LtcAdcStorage adc_storage;
+  CurrentSenseCalibrationData data;
+  CurrentSenseValue value;
+  int32_t offset;
+  bool offset_flag;
+  CurrentSenseCallback callback;
+  void *context;
+} CurrentSenseStorage;
+
+// Initialize the current sense module. Requires |data| to be calibrated beforehand.
+StatusCode current_sense_init(CurrentSenseStorage *storage, const CurrentSenseCalibrationData data,
+                              const LtcAdcSettings *settings);
+
+// Register a callback to run when new data is available
+StatusCode current_sense_register_callback(CurrentSenseStorage *storage,
+                                           CurrentSenseCallback callback, void *context);
+
+// Returns current value in microamps
+StatusCode current_sense_get_value(CurrentSenseStorage *storage, int32_t *current);
```

Comment:
Does this have a timeout or something?

Reply:
The most recent value is recorded in the storage, so there isn't any need to wait

Reply:
Yes, but what happens if the most recent value is from hours ago? Imagine if the current sensor gets disconnected due to a bad crimp.